### PR TITLE
fix: use native Node URL parser

### PR DESCRIPTION
**Pull request by Jolg42**
Remove https://github.com/unshiftio/url-parse

Use native URL https://nodejs.org/docs/latest-v10.x/api/url.html

Related https://github.com/prisma/prisma/issues/6362

### CODE REVIEW

File: src/packages/sdk/src/__tests__/__snapshots__/convertCredentials.test.ts.snap [hunk 1]
```diff
@@ -126,23 +126,59 @@ Object {
...
-  "host": undefined,
+  "host": "localhost",
   "password": undefined,
   "port": undefined,
   "schema": undefined,
   "socket": "/private/tmp/mysql.sock",
   "ssl": false,
   "type": "mysql",
-  "uri": "mysql://root@/db?socket=/private/tmp/mysql.sock",
+  "uri": "mysql://root@localhost/db?socket=/private/tmp/mysql.sock",
```
Comment (Jolg42) [inline src/packages/sdk/src/__tests__/__snapshots__/convertCredentials.test.ts.snap:140]: The downside of this approach is that this URL now errors like

```
> new URL('mysql://root@/db?socket=/private/tmp/mysql.sock')
Uncaught:
TypeError [ERR_INVALID_URL]: Invalid URL: mysql://root@/db?socket=/private/tmp/mysql.sock
    at onParseError (internal/url.js:258:9)
    at new URL (internal/url.js:334:5)
    at REPL19:1:1
    at Script.runInThisContext (vm.js:132:18)
    at REPLServer.defaultEval (repl.js:484:29)
    at bound (domain.js:430:14)
    at REPLServer.runBound [as eval] (domain.js:443:12)
    at REPLServer.onLine (repl.js:817:10)
    at REPLServer.emit (events.js:327:22)
    at REPLServer.EventEmitter.emit (domain.js:486:12) {
  input: 'mysql://root@/db?socket=/private/tmp/mysql.sock',
  code: 'ERR_INVALID_URL'
}
```

File: src/packages/sdk/src/__tests__/__snapshots__/convertCredentials.test.ts.snap [hunk 2]
```diff
@@ -311,23 +347,25 @@ Object {
...
+  "host": "localhost",
   "password": "prisma",
   "port": undefined,
   "schema": "public",
-  "socket": "/var/run/postgresql/",
+  "socket": undefined,
   "ssl": false,
   "type": "postgresql",
-  "uri": "postgresql://root:prisma@/prisma?host=/var/run/postgresql/",
+  "uri": "postgresql://root:prisma@localhost/prisma?host=/var/run/postgresql/",
```
Comment (Jolg42) [inline src/packages/sdk/src/__tests__/__snapshots__/convertCredentials.test.ts.snap:363]: The downside of this PR is that this URL now errors like (hostname is required)

```
> new URL('postgresql://root:prisma@/prisma?host=/var/run/postgresql/')
Uncaught:
TypeError [ERR_INVALID_URL]: Invalid URL: postgresql://root:prisma@/prisma?host=/var/run/postgresql/
    at onParseError (internal/url.js:258:9)
    at new URL (internal/url.js:334:5)
    at REPL17:1:1
    at Script.runInThisContext (vm.js:132:18)
    at REPLServer.defaultEval (repl.js:484:29)
    at bound (domain.js:430:14)
    at REPLServer.runBound [as eval] (domain.js:443:12)
    at REPLServer.onLine (repl.js:817:10)
    at REPLServer.emit (events.js:327:22)
    at REPLServer.EventEmitter.emit (domain.js:486:12) {
  input: 'postgresql://root:prisma@/prisma?host=/var/run/postgresql/',
  code: 'ERR_INVALID_URL'
}
```

File: src/packages/sdk/src/__tests__/__snapshots__/convertCredentials.test.ts.snap (hunk 1 above)
Comment (Jolg42) [reply src/packages/sdk/src/__tests__/__snapshots__/convertCredentials.test.ts.snap:140]: This error is now caught and replaced with in `uriToCredentials` function 
`Invalid data source URL. See https://www.prisma.io/docs/reference/database-reference/connection-urls`

File: src/packages/sdk/src/__tests__/__snapshots__/convertCredentials.test.ts.snap (hunk 2 above)
Comment (Jolg42) [reply src/packages/sdk/src/__tests__/__snapshots__/convertCredentials.test.ts.snap:363]: This error is now caught and replaced with in uriToCredentials function
`Invalid data source URL. See https://www.prisma.io/docs/reference/database-reference/connection-urls`

**Review by millsp — APPROVED** [review]

**Comment by Jolg42** [comment]
Notes:

So for an invalid URL (like without host)
- new URL will fire an error
- it’s caught
https://github.com/prisma/prisma/pull/6546/files#diff-3ec492a4fc94d6ef3fa70db09e0a44bdae1e81c2c22586e4c4ae1de3d60c8bb4R95-R111
And display will be not as good but CLI will not block

**Comment by Jolg42** [comment]
Notes:

So for an invalid URL (like without host)
- new URL will fire an error
- it’s caught
https://github.com/prisma/prisma/pull/6546/files#diff-3ec492a4fc94d6ef3fa70db09e0a44bdae1e81c2c22586e4c4ae1de3d60c8bb4R95-R111

So if parsing errors CLI will not block

**Pull request merged**